flight/Modules/Airspeed/airspeed.c

```diff
@@ -104,10 +104,10 @@ static int8_t airspeedADCPin=-1;
 // Private functions
 static void airspeedTask(void *parameters);
 void baro_airspeedGet(BaroAirspeedData *baroAirspeedData, uint32_t *lastSysTime, uint8_t airspeedSensorType, int8_t airspeedADCPin);
-static void AirspeedSettingsUpdatedCb(UAVObjEvent * ev);
+static void AirspeedSettingsUpdatedCb(UAVObjEvent * ev, void *ctx, void *obj, int len);
```

Comment:
I disagree with the wordier part. It's true that it is wordier for the caller, as it has to implement what you wrote above, but there is only one caller (the system module), whereas in you version, every implementation of a callback needs to add something like `(void) ev; (void) ctx; (void) obj; (void) len;` in order not to generate compiler warnings and the callback signature is wordier as well. In addition, if there are every any additions / modifications, it means that every callback function will need to be modified, whereas a struct could easily be extended.

It isn't clear to me what the use case of the "ctx" parameter is; can you give an example of it's intended purpose? What you said regarding compiler efficiency is true, but one could use `struct my_state *foo = ctx.ctx;` and after that it would be equally efficient to access  `foo->whatever`.